src/arch/svm/utils.ts

```diff
@@ -311,3 +348,280 @@ export const createDefaultTransaction = async (rpcClient: SVMProvider, signer: T
+): Promise<IAccountMeta<string>[]> {
+  const l1Usdc = EvmAddress.from(TOKEN_SYMBOLS_MAP.USDC.addresses[hubChainId]);
+  const l2Usdc = SvmAddress.from(
+    TOKEN_SYMBOLS_MAP.USDC.addresses[chainIsProd(hubChainId) ? CHAIN_IDs.SOLANA : CHAIN_IDs.SOLANA_DEVNET]
+  );
+
+  const [tokenMessengerPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [Buffer.from("token_messenger")],
+  });
+
+  const [tokenMinterPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [Buffer.from("token_minter")],
+  });
+
+  const [localTokenPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [Buffer.from("local_token"), bs58.decode(l2Usdc.toBase58())],
+  });
+
+  const [tokenMessengerEventAuthorityPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [Buffer.from("__event_authority")],
+  });
+
+  const [custodyTokenAccountPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [Buffer.from("custody"), bs58.decode(l2Usdc.toBase58())],
+  });
+
+  const tokenAccount = await getAssociatedTokenAddress(SvmAddress.from(svmSigner.keyPair.publicKey.toString()), l2Usdc);
+
+  // Define accounts dependent on deposit information.
+  const [tokenPairPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [
+      Buffer.from("token_pair"),
+      Buffer.from(String(message.sourceDomain)),
+      Buffer.from(l1Usdc.toBytes32().slice(2), "hex"),
+    ],
+  });
+
+  const [remoteTokenMessengerPda] = await getProgramDerivedAddress({
+    programAddress: tokenMessengerMinter,
+    seeds: [Buffer.from("remote_token_messenger"), Buffer.from(String(message.sourceDomain))],
+  });
+
+  return [
+    { address: tokenMessengerPda, role: AccountRole.READONLY },
+    { address: remoteTokenMessengerPda, role: AccountRole.READONLY },
+    { address: tokenMinterPda, role: AccountRole.WRITABLE },
+    { address: localTokenPda, role: AccountRole.WRITABLE },
+    { address: tokenPairPda, role: AccountRole.READONLY },
+    { address: tokenAccount, role: AccountRole.WRITABLE },
+    { address: custodyTokenAccountPda, role: AccountRole.WRITABLE },
+    { address: TOKEN_PROGRAM_ADDRESS, role: AccountRole.READONLY },
+    { address: tokenMessengerEventAuthorityPda, role: AccountRole.READONLY },
+    { address: tokenMessengerMinter, role: AccountRole.READONLY },
+  ];
+}
+
+/**
+ * Finalizes CCTP deposits and messages on Solana.
+ *
+ * @param attestedMessages The CCTP messages to Solana.
+ * @param signer A base signer to be converted into a Solana signer.
+ * @param simulate Whether to simulate the transaction.
+ * @param hubChainId The chain ID of the hub.
+ * @param solanaClient The Solana client.
+ * @returns A list of executed transaction signatures.
+ */
+
+export async function finalizeSvmMessages(
+  attestedMessages: AttestedCCTPMessage[],
```

Comment:
its bit unclear to me where do we perform the filtering of `attestedMessages` - is that the responsibility of the caller? cause this function expects this to be either tokenless message targeted to `svm_spoke` or receiving tokens to the `vault`